remotes/docker/resolver.go

```diff
@@ -489,7 +498,8 @@ func (r *dockerBase) fetchTokenWithOAuth(ctx context.Context, to tokenOptions) (
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode == 405 && r.username != "" {
+	// GCR returns 404 for POST /v2/token
```

Comment:
Thank you for adding the comment. Maybe clarify the comment here to mention that this condition is for registries that have not implemented the `POST` endpoint. Then call out the specific deviations such as a method not allowed when a username is given or the 404 case. I would prefer to stay clear of calling out specific registry operators in the code. You can remove or include my snarky comment about registries below along with that, as long as it is not targeting anyone 😛

Reply:
+1 on removing snarky comment ;)